src/renderer/dx/CustomTextLayout.cpp

```diff
@@ -137,10 +137,16 @@ try
 {
     const auto drawingContext = static_cast<const DrawingContext*>(clientDrawingContext);
 
-    const DWRITE_FONT_WEIGHT weight = _fontRenderData->DefaultFontWeight();
+    DWRITE_FONT_WEIGHT weight = _fontRenderData->DefaultFontWeight();
     DWRITE_FONT_STYLE style = _fontRenderData->DefaultFontStyle();
     const DWRITE_FONT_STRETCH stretch = _fontRenderData->DefaultFontStretch();
 
+    if (drawingContext->useBoldFont)
+    {
+        // TODO: "relative" bold?
+        weight = DWRITE_FONT_WEIGHT_BOLD;

```

Comment:
You could write something like this:
```cpp
weight = (weight * DWRITE_FONT_WEIGHT_BOLD) / DWRITE_FONT_WEIGHT_NORMAL;
```

Reply:
Yeah I could. Just I'm not sure if it's correct. Or is there a "correct" way to do this.

Reply:
That's equivalent to what I suggested above (that's there the 1.75 came from), but today I suspect that font weight might not be perceptually linear. It'd work as a first pass, and clearly getting the rest of the framework in-place is important.

It also needs to cap at 999, so it's really 
```
min(999,  (weight * DWRITE_FONT_WEIGHT_BOLD) / DWRITE_FONT_WEIGHT_NORMAL)
```

If there's any background on the CSS font-boldening algorithm (i.e. how they determined it), it might give a better approach that works for variable-weight fonts as well. They might however have just been "given these four weights, just move up one to make it bold", which is probably not a good result for variable-width fonts.

A totally alternative approach is to let the user specify through config, and default to `DWRITE_FONT_WEIGHT_BOLD`, so if the user overrides the normal font weight, they either also override the bold font weight, or accept that they might have made their default font _bolder_ than their bold font.

Actually, that seems like the best long-term result, and needs less magic, but I guess we need _some_ magic in the meantime?

Reply:
It isn't unreasonable to "let the user specify through config", but it's certainly harder than what I intended for this PR. That would also require modification in the Settings UI. Maybe we can consider it in the long-term plan.

> can we live with hard-coded "bold is always 700" in the meantime

I'm perfectly OK with it, TBH. Just like I'm OK with Quake mode without animations. But I'm worried that some other folks in the community might disagree.

Reply:
I agree that "exposing the user config" shouldn't block this PR. We'll how those who must suffer the slings and arrows of outraged users feel about it. ^_^

Reply:
I'm gonna say lets file this as a follow up discussion, and ship as-is for now.